lib/dialects/abstract/query-generator.js

```diff
@@ -1789,18 +1789,25 @@ var QueryGenerator = {
       }
     }
 
-    if (value) {
-      if (value.$or || value.$and) {
-        binding = value.$or ? ' OR ' : ' AND ';
+    if (value && (value.$or || value.$and)) {
+      binding = value.$or ? ' OR ' : ' AND ';
+      value = value.$or || value.$and;
 
-        value = (value.$or || value.$and).map(function (_value) {
-          return self.whereItemQuery(key, _value, options);
-        }).filter(function (item) {
-          return item && item.length;
+      if (_.isPlainObject(value)) {
+        var where = [];
+        _.forOwn(value, function (_value, key) {
+          where.push(_.zipObject([key], [_value]));
         });
```

Comment:
Pretty clever use of zip here to assign a variable as key without creating the object first. I like it :-)